### PR TITLE
Separate VM signature verification from execution

### PR DESCRIPTION
**Pull request by hmoog**
...

### CODE REVIEW

File: transaction.go [hunk 1]
```diff
@@ -53,24 +57,115 @@ type (
...
+	if err != nil {
+		return TransactionID{}, ierrors.Errorf("can't compute essence bytes: %w", err)
+	}
+
+	outputBytes, err := t.API.Encode(t.Outputs)
+	if err != nil {
+		return TransactionID{}, ierrors.Errorf("can't compute unlock bytes: %w", err)
+	}
+
+	return SignedTransactionIDFromData(t.CreationSlot, byteutils.ConcatBytes(essenceBytes, outputBytes)), nil
```
Comment (alexsporn) [inline transaction.go]: Why `SignedTransactionIDFromData`?

File: block.go [hunk 2]
```diff
@@ -486,7 +486,7 @@ func (b *BasicBlock) syntacticallyValidate(protocolBlock *ProtocolBlock) error {
 			return ierrors.Wrapf(ErrTransactionCreationSlotTooRecent, "block at slot %d with commitment input to slot %d", blockSlot, tx.Transaction.CreationSlot)
 		}
 
-		if cInput := tx.CommitmentInput(); cInput != nil {
+		if cInput := tx.Transaction.CommitmentInput(); cInput != nil {
```
Comment (piotrm50) [inline block.go]: rename tx to signed transaction

File: output.go [hunk 3]
```diff
@@ -124,6 +124,15 @@ func storageScoreOffsetOutput(rentStruct *RentStructure) StorageScore {
 // out of the referenced TransactionID and the output's index.
 type OutputID [OutputIDLength]byte
 
+func NewOutputID(slotIdentifier SlotIdentifier, index uint16) OutputID {
```
Comment (piotrm50) [inline output.go]: Unused, remove it

File: signed_transaction_test.go [hunk 4]
```diff
@@ -233,16 +240,16 @@ func TestTransaction_InputTypes(t *testing.T) {
 			}),
 		))
 
-	utxoInputs, err := transaction.Inputs()
+	utxoInputs, err := transaction.Transaction.Inputs()
```
Comment (piotrm50) [inline signed_transaction_test.go]: rename to signedTransaction

File: transaction.go [hunk 5]
```diff
@@ -146,3 +255,8 @@ func (t *Transaction) WorkScore(workScoreStructure *WorkScoreStructure) (WorkSco
 
 	return workscoreTransactionEssence.Add(workScoreOutputs)
 }
+
+func (t *Transaction) String() string {
+	// TODO: stringify for debugging purposes
+	return fmt.Sprintf("SignedTransaction[%v %v]", t.TransactionEssence, t.Outputs)
```
Comment (piotrm50) [inline transaction.go]:
```suggestion
	return fmt.Sprintf("Transaction[%v %v]", t.TransactionEssence, t.Outputs)
```

**Pull request merged**